### PR TITLE
remove pytest_indexd plugin registry

### PR DESCRIPTION
**Pull request by qqiao2024**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,6 +26,9 @@ cache:
 jobs:
   fast_finish: true
   include:
+    - name: test unit tests
+      script:
+        - tox -e unit_test
```
Comment (stilesj-uchicago) [inline .travis.yml:31]: When does testenv:parallel get called?

**Review by kulgan — COMMENTED** [review]
I gave this pytest plugin approach some thought and I really like the approach, but coupling it with indexclient may be causing some confusion about its usage and forces every project that makes use of indexclient to also use it. If you look for other pytest plugins, they are usually separate allowing users to decide whether to use it or not. I think moving this to a separate project with the following:

1. Separate concerns - which will clearly show that the pytest plugin and indexclient are two separate features one solely for testing purposes and the other for querying indexd server
2. Allow users to decide when to use the plugin and not automatic just by an indexclient dependency
3. An extension of point two, plan x makes use of their own copy of indexclient, maybe a pytest plugin for it is beneficial to them too. With this separation they can easily use it without depending on our own indexclient

**Comment by stilesj-uchicago** [comment]
> I gave this pytest plugin approach some thought and I really like the approach, but coupling it with indexclient may be causing some confusion about its usage and forces every project that makes use of indexclient to also use it. If you look for other pytest plugins, they are usually separate allowing users to decide whether to use it or not. I think moving this to a separate project with the following:
> 
>     1. Separate concerns - which will clearly show that the pytest plugin and indexclient are two separate features one solely for testing purposes and the other for querying indexd server
> 
>     2. Allow users to decide when to use the plugin and not automatic just by an indexclient dependency
> 
>     3. An extension of point two, plan x makes use of their own copy of indexclient, maybe a pytest plugin for it is beneficial to them too. With this separation they can easily use it without depending on our own indexclient

@kulgan , I think having the plugin as an optional install works fine for #1. I would hate to have yet another repo that is loosely coupled. The registration for the plugin has to be explicit in conftest.py (according to Qiao) to use it so it should not leak dependencies.  For number #2, including the plugin is not automatic so the user would have to explicitly include it in their dev-requirements.in.  For number #3, I think we have sufficiently diverged from plan x and I don't know how much we want to make sure we are supporting that effort too.

File: .travis.yml (hunk 1 above)
Comment (stilesj-uchicago) [reply .travis.yml:31]: @qiaouchicago you said you needed to remove this dead code. Please, let me know once you have done so. Thank you.

**Review by stilesj-uchicago — APPROVED** [review]

**Review by jiakf — APPROVED** [review]

**Review by kulgan — APPROVED** [review]

**Pull request merged**